### PR TITLE
Publish Dialog updated to use new components and styles 

### PR DESCRIPTION
**Pull request by Erin007**
We have too many button components and too many dialog components.  We should declutter our component library as it will easier to maintain and will increase stylistic consistency across the site. Notably we still use `LegacyButton` in a handful of places even though we say it is deprecated. 

One such use of `LegacyButton` is in the `PublishDialog`, which also uses the outdated `Dialog` component.  In an effort to chip away at the tech debt of maintaining a multitude of similar components, I updated `PublishDialog` to use the newer `BaseDialog` and `Button` components. In doing so, I also extended the functionality of `Button` to include a pending variation, which over the long term inches us closer to being able to fully deprecate not only `LegacyButton` but also `PendingButton`. 

BEFORE
screen shot 2018-06-25 at 5 38 54 pm

AFTER 
screen shot 2018-06-25 at 5 42 49 pm

Pending variant of ` Button`:
pendingbutton

### CODE REVIEW

File: apps/src/templates/publishDialog/PublishDialog.jsx [hunk 1]
```diff
@@ -1,12 +1,20 @@
...
+import DialogFooter from '../teacherDashboard/DialogFooter';
+import Button from '../Button';
 import i18n from '@cdo/locale';
 import { hidePublishDialog, publishProject } from './publishDialogRedux';
 
+const styles = {
+  dialog: {
+    paddingLeft: 20,
+    paddingRight: 20,
+    paddingBottom: 20
```
Comment (joshlory) [inline apps/src/templates/publishDialog/PublishDialog.jsx]: Tiny style nit: missing trailing comma.

File: apps/src/templates/Button.jsx [hunk 2]
```diff
@@ -178,7 +180,15 @@ class Button extends React.Component {
               style={{...styles.icon, ...iconStyle}}
             />
           }
-          {text}
+          {isPending && pendingText &&
```
Comment (joshlory) [inline apps/src/templates/Button.jsx:183]: Should the button also be marked `disabled` when pending?

File: apps/src/templates/publishDialog/PublishDialog.jsx [hunk 3]
```diff
@@ -42,29 +50,36 @@ class PublishDialog extends Component {
 
   render() {
     return (
-      <Dialog
-        title={i18n.publishToPublicGallery()}
+      <BaseDialog
         isOpen={this.props.isOpen}
         handleClose={this.close}
+        useUpdatedStyles
```
Comment (joshlory) [inline apps/src/templates/publishDialog/PublishDialog.jsx:56]: What does this do?

**Comment by joshlory** [comment]
Is this scenario covered by a UI test?  Circle failure look related.

File: apps/src/templates/publishDialog/PublishDialog.jsx (hunk 3 above)
Comment (davidsbailey) [reply apps/src/templates/publishDialog/PublishDialog.jsx:56]: https://reactjs.org/docs/jsx-in-depth.html#props-default-to-true

Comment (joshlory) [reply apps/src/templates/publishDialog/PublishDialog.jsx:56]: Sorry — what is the purpose of passing this prop?  I see where it is used, but the prop isn't documented.  Mostly surprised that we're still relying on global CSS, even in the "updated" version of the component.

https://github.com/code-dot-org/code-dot-org/blob/acd8a7f3474a7e505d4c66241fceb785f9c7f67d/apps/src/templates/BaseDialog.jsx#L93-L97

Comment (Erin007) [reply apps/src/templates/publishDialog/PublishDialog.jsx:56]: Without `useUpdatedStyles`
screen shot 2018-06-26 at 12 57 33 pm
The style differences are that without `useUpdatedStyles` we end up with the old style purple x in the corner to dismiss the dialog, the bottom corners are rounded and the width shrinks some.

Oddly, I also think using the new styles removes the `.modal-body` class that the UI tests are expecting to find.

Ideally, we probably want: 
- gray x to dismiss 
- wider width 
- 4 consistently rounded corners 
- modal-body class for all `BaseDialogs` 
- no reliance on global styles 
- remove both `useUpdatedStyles` and `useDeprecatedGlobalStyles` props in favor of uniform, consistent styling 

`BaseDialog` needs to be updated to new ES6 syntax anyway, so I'm going to check with Mark about what design we actually want and then update that component as follow up.

Comment (joshlory) [reply apps/src/templates/publishDialog/PublishDialog.jsx:56]: > remove both useUpdatedStyles and useDeprecatedGlobalStyles props in favor of uniform, consistent styling

🎉

Not blocking this PR of course, but that sounds like awesome follow up!

**Pull request merged**